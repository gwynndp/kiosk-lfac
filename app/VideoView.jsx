import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import MainNav from './MainNav';
import Sidebar from './Sidebar';

const StyledVideoView = styled.div`
  background-color: white;
  padding: 20px 50px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  width: 70%;
`;

const Button = styled.button`
  background-color: #ecd72b;
  border-radius: 6px;
  width: 40%;
  padding: 15px;
  margin: 20px;
  font-size: 20px;
  font-weight: 600;
  text-transform: uppercase;
  border: none;
`;

function handlePrevNext(vidId, listLength, updateCurrentVideo, diff) {
  const currVid = vidId;
  let newIndex = 0;
  if (diff > 0) {
    newVid = currVid < listLength - 1 ? currVid + 1 : 0;
  } else {
    newVid = currVid > 0 ? currVid - 1 : listLength - 1;
  }
  updateCurrentVideo(newIndex);
}

export default function VideoView({ video, listLength, updateCurrentVideo }) {
  return (
    <React.Fragment>
      <MainNav gallery='false' />
      <StyledVideoView>
        <h1 style={{ fontSize: '42px' }}>{video.name}</h1>
        <video
          controls
          src={video.src}
          poster={video.img}
          style={{
            //width: `800px`,
            flexShrink: '1',
            flexGrow: '1',
            width: '100%',
          }}
          autoPlay
          allowFullScreen
          title={`Video interview of ${video.name}`}
        />
        <div
          style={{
            display: 'flex',
            flexDirection: 'row',
            justifyContent: 'space-around',
            alignSelf: 'stretch',
            margin: '20px',
          }}
        >
          <Button
            onClick={() =>
              handlePrevNext(video.id, listLength, updateCurrentVideo, -1)
            }
          >
            Previous
          </Button>
          <Button
            onClick={() =>
              handlePrevNext(video.id, listLength, updateCurrentVideo, 1)
            }
          >
            Next
          </Button>
        </div>
      </StyledVideoView>
      <Sidebar />
    </React.Fragment>
  );
}

VideoView.propTypes = {
  video: PropTypes.object.isRequired,
  listLength: PropTypes.number.isRequired,
  updateCurrentVideo: PropTypes.func.isRequired,
};
