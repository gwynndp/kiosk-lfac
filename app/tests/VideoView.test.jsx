import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import VideoView from "../VideoView";

test("should render a VideoView", () => {
  const props = {
    currentVideo: {
      id: 4,
      category: "people",
      name: "Ashley Sanchez",
      src: "videos/Ashley-Sanchez.mp4",
      img: "images/videos/Ashley-Sanchez.jpg"
    },
    listLength: 5,
    updateCurrentVideo: () => {}
  };

  const component = renderer.create(
    <Router>
      <VideoView
        video={props.currentVideo}
        listLength={props.listLength}
        updateCurrentVideo={props.updateCurrentVideo}
      />
    </Router>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

// should render a MainNav
// should render a Video
// should render buttons for Prev and Next

// should render prev video when prev button clicked
// should render next vidwo when next button clicked
