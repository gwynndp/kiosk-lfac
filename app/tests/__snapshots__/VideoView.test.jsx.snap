// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a VideoView 1`] = `
Array [
  <div
    className="sc-AykKC duquAX"
  >
    <a
      className="sc-AykKD dWbVqE"
      exact="true"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="home"
        src="images/kiosk_home.png"
        width="100px"
      />
    </a>
  </div>,
  <div
    className="sc-AykKH eNnoAM"
  >
    <h1
      style={
        Object {
          "fontSize": "42px",
        }
      }
    >
      Ashley Sanchez
    </h1>
    <video
      allowFullScreen={true}
      autoPlay={true}
      controls={true}
      poster="images/videos/Ashley-Sanchez.jpg"
      src="videos/Ashley-Sanchez.mp4"
      style={
        Object {
          "flexGrow": "1",
          "flexShrink": "1",
          "width": "100%",
        }
      }
      title="Video interview of Ashley Sanchez"
    />
    <div
      style={
        Object {
          "alignSelf": "stretch",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "margin": "20px",
        }
      }
    >
      <button
        className="sc-AykKI fdsJZZ"
        onClick={[Function]}
      >
        Previous
      </button>
      <button
        className="sc-AykKI fdsJZZ"
        onClick={[Function]}
      >
        Next
      </button>
    </div>
  </div>,
  <div
    className="sc-AykKF cvNFcn"
  >
    <div
      id="logos-wrapper"
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": "0 10px 20px 0",
        }
      }
    >
      <img
        alt="Cloverdale City logo"
        src="images/cloverdale-city-logo.png"
        width="100%"
      />
      <img
        alt="Cloverdale High School logo"
        src="images/cloverdale-hs-logo.png"
        width="100%"
      />
      <img
        alt="Listening For a Change logo"
        src="images/lfac_logo_black.gif"
        width="100%"
      />
    </div>
    <a
      className="sc-AykKG esZqhL"
      href="/lfac"
      onClick={[Function]}
    >
      <span>
        Listening for a Change
      </span>
    </a>
    <a
      className="sc-AykKG esZqhL"
      href="/schools"
      onClick={[Function]}
    >
      <span>
        Schools
      </span>
    </a>
    <a
      className="sc-AykKG esZqhL"
      href="/fires"
      onClick={[Function]}
    >
      <span>
        Fire Stories
      </span>
    </a>
    <a
      className="sc-AykKG esZqhL"
      href="/jackdemayo"
      onClick={[Function]}
    >
      <span>
        Remembering Jack DeMayo
      </span>
    </a>
    <a
      className="sc-AykKG esZqhL"
      href="/vom"
      onClick={[Function]}
    >
      <span>
        Valley of the Moon
      </span>
    </a>
  </div>,
]
`;
